tests/integration/kubernetes/k8s-guest-pull-image-signature.bats

```diff
@@ -0,0 +1,179 @@
+        "io.containerd.cri.runtime-handler" \
+        "kata-${KATA_HYPERVISOR}"
+    set_metadata_annotation "${kata_pod_with_signed_image}" \
+        "${kernel_params_annotation}" \
+        "${kernel_params_value}"
+
+    add_allow_all_policy_to_yaml "${kata_pod_with_signed_image}"
+}
+
+@test "Create a pod from an unsigned image, on an insecureAcceptAnything registry works" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_UNPROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+@test "Create a pod from an unsigned image, on a 'restricted registry' is rejected" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_PROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    assert_pod_fail "${kata_pod_with_signed_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" "Security validate failed: Validate image failed: Cannot pull manifest of ghcr.io/confidential-containers/test-container-image-rs:sha256-28e01ab32c9dbcbaae96cf0d5b472f22e231d9e603811857b295e61197e40a9b.sig"
+}
+
+@test "Create a pod from a signed image, on a 'restricted registry' is successful" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${COSIGN_SIGNED_PROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+@test "Create a pod from a signed image, on a 'restricted registry', but with the wrong key is rejected" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${COSIGNED_SIGNED_PROTECTED_REGISTRY_WRONG_KEY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    assert_pod_fail "${kata_pod_with_signed_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" "Security validate failed: Validate image failed: [PublicKeyVerifier"
+}
+
+@test "Create a pod from an unsigned image, on a 'restricted registry' works if enable_signature_verfication is false" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_PROTECTED_REGISTRY_IMAGE}" "false"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+teardown() {
+    # if is_confidential_hardware; then
+    #     skip "Due to issues related to pull-image integration skip tests for ${KATA_HYPERVISOR}."
+    # fi
```

Comment:
We need to decide whether or not to run this on the TEE hardware, so lets run the e2e tests and make a call to delete this, or uncomment

Reply:
The very first features were tested on non-TEE first and after a while we enabled on TEE. Then we kind of changed the approach to run in all platforms (TEE and non-TEE) and IMHO it has been a little bit frustrating because TEE CI still fails a lot. So IMO we should skip TEE testing for a while; let's have the feature merged with qemu-coco-dev.

Reply:
In meanwhile TEE CI got stable so that running on there and non-TEE is fine.

Reply:
Yeah, I've updated the tests to run for all the coco runtime classes now